### ISSUE TITLE
Tagging: handling invalid and/or multiple entries by the tagging function

### ISSUE THREAD

**Issue by darrudi** [angular-ui/ui-select#485]
Feature Request regarding free tagging:
The user may enter a string which is considered invalid by the tagging function. So the tagging function should be able to return null upon receiving the input. I did this but a bunch of errors surfaced. My tagging function (an email client):
```
    $scope.uiSelectNewTag = function(email){
        email = email.trim();

        if (email.match(/^[a-z][a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/i)) {
            var newItem = {
                type: 2,
                email: email,
                key: email
            };
            return newItem;

        } else {
            return null;
        }
    };
```
Also the user may copy-paste a string (e.g. a list of emails separated by semicolons) into the ui-select and the tagging function may decide there are more than one item in the input. So I guess its a good idea to have the tagging function return an array (zero or more items) instead of a string.

**Comment by NJAldwin** [comment]
Regarding the second point: couldn't you solve that by using `tagging-tokens`?

I'm all for the main issue here, namely adding some sort of tag validation.

**Comment by darrudi** [comment]
Regarding `tagging-tokens`:
I doubt if copy-pasting data containing several `tagging-tokens` (`;` for example) into the ui-select input would trigger the `tagging` function several times. I did not give it a try. The current tagging function does not support other desirable functionality, i.e. it does not create a tag onBlur, so I could not actually use it. I ended up to customizing ui-select to implement the missing features.

**Comment by vdsabev** [comment]
@darrudi we have the exact same use case with emails:

1. Adding entries via the tagging feature
2. Copy-pasting what could be multiple entries
3. Adding the entry on input blur
4. Not adding invalid entries at all

Would you mind sharing how you customized the plugin to achieve this?

**Comment by darrudi** [comment]
@vdsabev: sorry for the delay. I guess you'd be better off with ngTagsInput as my customizations were somehow hackish.

**Comment by LeleDev** [comment]
+1 for this feature.

It would be really really useful to return a null with the tagging function when you don't want to add a not-valid field

**Comment by dimirc** [comment]
Seems to be resolved with #678 and #664
